dataavailability/interfaces.go

```diff
@@ -0,0 +1,25 @@
+package dataavailability
+
+import (
+	"context"
+
+	"github.com/0xPolygonHermez/zkevm-node/state"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/jackc/pgx/v4"
+)
+
+type stateInterface interface {
+	GetBatchL2DataByNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) ([]byte, error)
+	GetBatchByNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) (*state.Batch, error)
+}
+
+// DABackender is the interface for a DA backend
+type DABackender interface {
+	// Init initializes the DABackend
+	Init() error
+	// GetData retrieve the data of a batch from the DA backend. The returned data must be the pre-image of the hash
+	GetData(batchNum uint64, hash common.Hash) ([]byte, error)
+	// PostSequence sends the sequence data to the data availability backend, and returns the dataAvailabilityMessage
+	// as expected by the contract
+	PostSequence(ctx context.Context, batchesData [][]byte) ([]byte, error)
+}
```

Comment:
WDYT about such interface segregation, because if I'm not mistaken, it seems that sometimes you need to implement just the `BatchDataProvider` (reading batches) and sometimes only `SequenceSender` (posting batches, e.g. here https://github.com/0xPolygon/cdk-validium-node/blob/9d186be306740f2fa9e1dcc4186124baa65c6d77/sequencesender/interfaces.go#L42). Does it make sense, or it is an overkill?

```suggestion
// BatchDataProvider is used to retrieve batch data
type BatchDataProvider interface {
	// GetData retrieve the data of a batch from the DA backend. The returned data must be the pre-image of the hash
	GetData(batchNum uint64, hash common.Hash) ([]byte, error)
}

// SequenceSender is used to send provided sequence of batches
type SequenceSender interface {
	// PostSequence sends the sequence data to the data availability backend, and returns the dataAvailabilityMessage
	// as expected by the contract
	PostSequence(ctx context.Context, batchesData [][]byte) ([]byte, error)
}

type DABackender interface {
	BatchDataProvider
	SequenceSender
	// Init initializes the DABackend
	Init() error
}
```